tests/components/mailbox/test_init.py

```diff
@@ -0,0 +1,158 @@
+"""The tests for the mailbox component."""
+import json
+import requests
+from hashlib import sha1
+
+from homeassistant.setup import setup_component
+import homeassistant.components.mailbox as mailbox
+import homeassistant.components.http as http
+
+from tests.common import (
+    get_test_home_assistant, get_test_instance_port, assert_setup_component)
+
+
+class TestSetupMailbox(object):
+    """Verify mailbox creation."""
+
+    def setup_method(self):
+        """Initialize mailbox platform."""
+        self.hass = get_test_home_assistant()
+
+    def teardown_method(self):
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    def test_setup_component(self):
+        """Initialize demo platform on mailbox component."""
+        config = {
+            mailbox.DOMAIN: {
+                'platform': 'demo'
+            }
+        }
+
+        with assert_setup_component(1, mailbox.DOMAIN):
+            setup_component(self.hass, mailbox.DOMAIN, config)
+
+
+class TestMailbox(object):
+    """Test class for mailbox."""
+
+    def setup_method(self):
+        """Initialize platform."""
+        self.hass = get_test_home_assistant()
+
+        setup_component(
+            self.hass, http.DOMAIN,
+            {http.DOMAIN: {http.CONF_SERVER_PORT: get_test_instance_port()}})
+
+        config = {
+            mailbox.DOMAIN: {
+                'platform': 'demo'
+            }
+        }
+
+        setup_component(self.hass, mailbox.DOMAIN, config)
```

Comment:
You always want to `assert` the result of `setup_component`. That way it will fail the test early if setup fails.